### PR TITLE
GIX-996: SNS Transactions

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Add method to get the transactions of an account.

# Changes

* Add SnsIndexCanister with "getTransactions" method.

# Tests

* Test for the new sns canister.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@peterpeterparker @mstrasinskis please review

File: packages/sns/src/sns-index.canister.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        options,
+        idlFactory,
+        certifiedIdlFactory,
+      });
+
+    return new SnsIndexCanister(canisterId, service, certifiedService);
+  }
+
+  /**
+   * Get the state of the swap
```
Comment (peterpeterparker) [inline packages/sns/src/sns-index.canister.ts:26]: this comment is correct?

File: packages/sns/src/sns-index.canister.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+   *
+   * Always certified.
+   * `get_account_transactions` needs to be called with an update to get the transactios from the ledger canister.
```
Comment (peterpeterparker) [inline packages/sns/src/sns-index.canister.ts:29]: maybe add a comment why it needs to be certified?

File: packages/sns/src/sns-index.canister.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+   */
+  getTransactions = async (
+    params: GetAccountTransactionsParams
+  ): Promise<GetTransactions> => {
+    const response = await this.caller({
+      certified: true,
+    }).get_account_transactions(toGetTransactionsArgs(params));
+
+    if ("Err" in response) {
+      throw new Error(response.Err.message);
```
Comment (peterpeterparker) [inline packages/sns/src/sns-index.canister.ts:39]: to follow the pattern of the ledger and governance should we return a `SnsIndexError`?

**Comment by peterpeterparker** [comment]
@lmuntaner shouldn't this new "index" canister be added to the `sns.wrapper.ts` and `sns.ts` or you will do that in another PR or it shouldn't?

File: packages/sns/src/sns-index.canister.ts (hunk 1 above)
Comment (lmuntaner) [reply packages/sns/src/sns-index.canister.ts:26]: Nops, I copy paster and forgot to change this part but not below.

File: packages/sns/src/sns-index.canister.ts (hunk 2 above)
Comment (lmuntaner) [reply packages/sns/src/sns-index.canister.ts:29]: That's what the comment is saying, haha. But I can see it's not clear at all.

File: packages/sns/src/sns-index.canister.ts (hunk 3 above)
Comment (lmuntaner) [reply packages/sns/src/sns-index.canister.ts:39]: ok

**Comment by lmuntaner** [comment]
> @lmuntaner shouldn't this new "index" canister be added to the `sns.wrapper.ts` and `sns.ts` or you will do that in another PR or it shouldn't?

In another PR.

They have just added it to the root canister., but we can't test it yet. If I add it now, the wrapper won't be working with our SNS projects.

**Comment by lmuntaner** [comment]
@peterpeterparker ready for another review

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

**Pull request merged**